File: DotNetGotchas/CSharp/CopyingObjects/Modified8/Copy/SmarterBrain.cs
//SmarterBrain.cs
using System;

namespace Copy
{
	public class SmarterBrain : Brain
	{
		public SmarterBrain()
		{
		}

		protected SmarterBrain(SmarterBrain another) 
			: base(another)
		{
		}

		public override object Clone()
		{
			return new SmarterBrain(this);
		}

	}
}

File: DotNetGotchas/VBDN/CopyingObjects/Modified7/Copy/SmarterBrain.vb
'SmarterBrain.vb

Public Class SmarterBrain
    Inherits Brain

    Public Sub New()

    End Sub

    Protected Sub New(ByVal another As SmarterBrain)
        MyBase.New(another)
    End Sub


    Public Overrides Function Clone() As Object
        Return New SmarterBrain(Me)
    End Function
End Class
